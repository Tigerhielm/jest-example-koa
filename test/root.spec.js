const app = require('../server/index');
const request = require('supertest');

describe('root route', () => {
  test('with item-level assertions', async () => {
    const response = await request(app.callback()).get('/');

    expect(response.status).toEqual(200);
    expect(response.type).toEqual('application/json');
    expect(response.body.data).toEqual('Sending some JSON');
    expect(Object.keys(response.body.person)).toEqual(
      expect.arrayContaining(['name', 'lastname', 'role', 'age'])
    );
  });

  test('with object equality', async () => {
    const response = await request(app.callback()).get('/');

    expect(response.body).toEqual(
      expect.objectContaining({
        person: {
          name: expect.anything(),
          lastname: expect.any(String),
          role: expect.stringMatching(/^Brewery/),
          age: expect.any(Number)
        }
      })
    );
  });

  test('with object equality', async () => {
    const response = await request(app.callback()).get('/');

    expect(response.body).toMatchSnapshot();
  });
});
