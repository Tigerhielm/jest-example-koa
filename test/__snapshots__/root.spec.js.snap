// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root route with object equality 1`] = `
Object {
  "data": "Sending some JSON",
  "person": Object {
    "age": 42,
    "lastname": "VanÄ›k",
    "name": "Ferdinand",
    "role": "Brewery worker",
  },
}
`;
